ckanext/datastore/writer.py

```diff
@@ -11,6 +12,14 @@
 
 BOM = "\N{bom}"
 
+# (canada fork only): conform to XML stadards for element names
+# TODO: upstream contrib
+xml_element_name_rules = [
+    (re.compile(r'^([0-9]+xml|xml[0-9]+|xml|^[0-9]+)', re.I), ''),
+    (re.compile(r'\ +'), '_'),
+    (re.compile(r'(?:(?![a-z]|[A-Z]|[0-9]|-|_|\.).)'), ''),
```

Comment:
Are accented letters not allowed? You could use `\w` here instead

Reply:
@wardi https://www.w3.org/TR/xml/#NT-NameStartChar see `[4a] NameChar`. Idk how to do hex ranges in regex.

Reply:
```
Document authors are encouraged to use names which are meaningful words or combinations of words in natural languages, and to avoid symbolic or white space characters in names. Note that COLON, HYPHEN-MINUS, FULL STOP (period), LOW LINE (underscore), and MIDDLE DOT are explicitly permitted.

The ASCII symbols and punctuation marks, along with a fairly large group of Unicode symbol characters, are excluded from names because they are more useful as delimiters in contexts where XML names are used outside XML documents; providing this group gives those contexts hard guarantees about what cannot be part of an XML name. The character #x037E, GREEK QUESTION MARK, is excluded because when normalized it becomes a semicolon, which could change the meaning of entity references.
```

@wardi I think reading through this, we can just add into our regex to allow accented chars with `[à-ü]` and not worry about any crazier unicode chars.

Reply:
The ones included in `\w` are only "word characters" and don't include any punctuation like `'\u037e'` using a range will exclude lots of characters useful in other languages